mne/inverse_sparse/mxne_optim.py

```diff
@@ -339,30 +336,27 @@ def _mixed_norm_solver_bcd(M, G, alpha, lipschitz_constant, maxit=200,
 
     alpha_lc = alpha / lipschitz_constant
 
-    for i in range(maxit):
-        for j in range(n_positions):
-            idx = slice(j * n_orient, (j + 1) * n_orient)
-
-            G_j = G[:, idx]
-            X_j = X[idx]
-
-            X_j_new = np.dot(G_j.T, R) / lipschitz_constant[j]
+    # I understand it is not very clear to call gemm and gemm2 here
+    # however it is faster to call it only once
+    gemm = linalg.get_blas_funcs("gemm", [G[:, 0:n_orient].T, R])
+    gemm2 = linalg.get_blas_funcs("gemm", [G[:, 0:n_orient],
+                                  X[0:n_orient, :]])
```

Comment:
is it like gemm_C and next gemm_F ?
it will help to understand why you need 2 functions.

Reply:
No, this is the same gemm function, applied on fortran arrays, but on matrices of different size.

Reply:
Can you dig into `get_blas_funcs`? IIRC all it uses about the arrays is the dtypes and the contiguousness, not the sizes

Reply:
OK so there is no need to call two functions, I will change this